File: app/src/main/java/com/past/music/fragment/QuickControlsFragment.java
package com.past.music.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.facebook.drawee.view.SimpleDraweeView;
import com.neu.gaojin.MyOkHttpClient;
import com.neu.gaojin.response.BaseSuccessCallback;
import com.past.music.MyApplication;
import com.past.music.activity.PlayMusicActivity;
import com.past.music.api.AvatarRequest;
import com.past.music.api.AvatarResponse;
import com.past.music.dialog.MusicQueueFragment;
import com.past.music.log.MyLog;
import com.past.music.pastmusic.R;
import com.past.music.service.MusicPlayer;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * =======================================================
 * 作者：gaojin
 * 日期：2017/3/3 下午4:47
 * 描述：底部音乐快捷控制栏
 * 备注：备注：Copyright © 2010-2017. gaojin All rights reserved.
 * =======================================================
 */
public class QuickControlsFragment extends BaseFragment {

    private static final String TAG = "QuickControlsFragment";

    @BindView(R.id.playbar_img)
    SimpleDraweeView mAlbum;

    @BindView(R.id.rl_layout)
    RelativeLayout relativeLayout = null;

    @BindView(R.id.playbar_info)
    TextView mPlaybarInfo;

    @BindView(R.id.playbar_singer)
    TextView mPlaybarSinger;

    @BindView(R.id.play_list)
    ImageView mPlayList;

    @BindView(R.id.control)
    ImageView mControl;

    @BindView(R.id.play_next)
    ImageView mPlayNext;

    @OnClick(R.id.rl_layout)
    void rlLayout() {
        Intent intent = new Intent(getActivity(), PlayMusicActivity.class);
        startActivity(intent);
        getActivity().overridePendingTransition(R.anim.push_down_in, R.anim.push_up_out);
    }

    @OnClick(R.id.play_list)
    void playList() {
        MusicQueueFragment.newInstance().show(getFragmentManager(), "music_queue");
    }

    @OnClick(R.id.control)
    void control() {
        MusicPlayer.playOrPause();
        MyLog.i(TAG, "control");
    }

    @OnClick(R.id.play_next)
    void playNext() {
        MusicPlayer.nextPlay();
    }


    public static QuickControlsFragment newInstance() {
        return new QuickControlsFragment();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.view_quick_controls, container, false);
        ButterKnife.bind(this, rootView);
        return rootView;
    }

    @Override
    public void onResume() {
        super.onResume();
        updateFragment();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
    }

    public void updateFragment() {
        mPlaybarInfo.setText(MusicPlayer.getTrackName());
        mPlaybarSinger.setText(MusicPlayer.getArtistName());
        if (MusicPlayer.getIsPlaying()) {
            mControl.setImageResource(R.drawable.playbar_btn_pause);
        } else {
            mControl.setImageResource(R.drawable.playbar_btn_play);
        }
        if (MusicPlayer.getArtistName() != null) {
            AvatarRequest avatarRequest = new AvatarRequest();
            avatarRequest.setArtist(MusicPlayer.getArtistName().replace(";", " "));
            if (MyApplication.dbService.query(MusicPlayer.getArtistName().replace(";", "")) == null) {
                MyOkHttpClient.getInstance(getContext()).sendNet(avatarRequest, new BaseSuccessCallback<AvatarResponse>() {
                    @Override
                    public void onSuccess(int statusCode, final AvatarResponse response) {
                        MyLog.i("onSuccess", statusCode + "");
                        MyApplication.dbService.insert(MusicPlayer.getArtistName().replace(";", ""), response.getArtist().getImage().get(2).get_$Text112());
                        mAlbum.setImageURI(response.getArtist().getImage().get(2).get_$Text112());
                    }
                });
            } else {
                mAlbum.setImageURI(MyApplication.dbService.query(MusicPlayer.getArtistName().replace(";", "")));
            }
        }
    }

    @Override
    public void updatePlayInfo() {
        updateFragment();
    }
}

File: app/src/main/java/com/se/music/main/QuickControlsFragment.kt
package com.se.music.main

import android.animation.ObjectAnimator
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.LinearInterpolator
import android.widget.ImageView
import android.widget.TextView
import com.se.music.R
import com.se.music.activity.PlayingActivity
import com.se.music.base.BaseFragment
import com.se.music.base.Null
import com.se.music.entity.AlbumEntity
import com.se.music.service.MusicPlayer
import com.se.music.utils.isContentEmpty
import com.se.music.utils.loadAlbumPic


/**
 *Author: gaojin
 *Time: 2018/5/23 下午10:32
 */

class QuickControlsFragment : BaseFragment(), View.OnClickListener {

    companion object {
        fun newInstance(): QuickControlsFragment {
            return QuickControlsFragment()
        }
    }

    private lateinit var album: ImageView
    private lateinit var playBarSongName: TextView
    private lateinit var playBarSinger: TextView
    private lateinit var playList: ImageView
    private lateinit var control: ImageView
    private lateinit var playNext: ImageView
    private lateinit var mLogan: TextView
    private lateinit var circleAnim: ObjectAnimator

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        val root = inflater.inflate(R.layout.view_quick_controls, container, false)
        album = root.findViewById(R.id.play_bar_img)
        playBarSongName = root.findViewById(R.id.play_bar_song_name)
        playBarSinger = root.findViewById(R.id.play_bar_singer)
        playList = root.findViewById(R.id.play_list)
        control = root.findViewById(R.id.control)
        playNext = root.findViewById(R.id.play_next)
        mLogan = root.findViewById(R.id.music_logan)

        val albumCenterPoint = resources.getDimension(R.dimen.bottom_fragment_album_size) / 2
        album.pivotX = albumCenterPoint
        album.pivotY = albumCenterPoint

        circleAnim = ObjectAnimator.ofFloat(album, "rotation", 0.toFloat(), 360.toFloat())
        circleAnim.interpolator = LinearInterpolator()
        circleAnim.repeatCount = -1
        circleAnim.duration = 12000
        circleAnim.start()

        root.setOnClickListener(this)
        root.elevation = 100.toFloat()

        playList.setOnClickListener(this)
        control.setOnClickListener(this)
        playNext.setOnClickListener(this)
        return root
    }

    override fun onResume() {
        super.onResume()
        updateFragment()
    }

    override fun onClick(v: View) {
        when {
            v.id == R.id.play_list -> {
            }
            v.id == R.id.control -> MusicPlayer.playOrPause()
            v.id == R.id.play_next -> MusicPlayer.nextPlay()
            else -> {
                val intent = Intent(activity, PlayingActivity::class.java)
                startActivity(intent)
                activity!!.overridePendingTransition(R.anim.push_down_in, R.anim.push_up_out)
            }
        }
    }

    override fun updatePlayInfo() {
        updateFragment()
    }

    private fun updateFragment() {
        val albumEntity = AlbumEntity(MusicPlayer.getCurrentAlbumId()
                , MusicPlayer.getAlbumName()
                , 0
                , MusicPlayer.getArtistName()
                , Null)
        albumEntity.imageUrl = MusicPlayer.getAlbumPic()

        album.loadAlbumPic(context!!, albumEntity, loaderManager, R.drawable.player_albumcover_default)

        if (MusicPlayer.getTrackName().isContentEmpty() && MusicPlayer.getArtistName().isContentEmpty()) {
            playBarSongName.visibility = View.GONE
            playBarSinger.visibility = View.GONE
            album.visibility = View.GONE
            mLogan.visibility = View.VISIBLE
        } else {
            playBarSongName.visibility = View.VISIBLE
            playBarSinger.visibility = View.VISIBLE
            album.visibility = View.VISIBLE
            mLogan.visibility = View.GONE
            playBarSongName.text = MusicPlayer.getTrackName()
            playBarSinger.text = MusicPlayer.getArtistName()
        }

        if (MusicPlayer.isPlaying()) {
            control.setImageResource(R.drawable.playbar_btn_pause)
            circleAnim.resume()
        } else {
            control.setImageResource(R.drawable.playbar_btn_play)
            circleAnim.pause()
        }
    }
}